frappe_state_management/classes/fsm_document.py

```diff
@@ -50,17 +50,22 @@ def apply_update_request(self, update_request: UpdateRequest) -> None:
     try:
       if self.update_request.status in ('Approved', 'Pending'):
         method_call = None
+        revert_data = None
         if self.update_request.custom_call:
           if '.' in self.update_request.custom_call:
             method_call = frappe.get_attr(self.update_request.custom_call)
           else:
             method_call = getattr(self, self.update_request.custom_call, None)
+        elif frappe.get_hooks('fsm_fields', {}).get(self.doctype, {}).get(self.update_request.docfield):
+          revert_data = frappe.call(frappe.get_hooks('fsm_fields', {}).get(self.doctype, {}).get(self.update_request.docfield)[-1], self)
```

Comment:
Why the index at `-1`?

Reply:
because it's return list. it's from frappe